File: credential/credential.go
package credential

import "sync"

type AliyunCred struct {
	AccessKeyID     string
	AccessKeySecret string
}

type ESXiCred struct {
	URL      string
	User     string
	Password string
}

type DeepSeekCred struct {
	APIKey string
}

type Store struct {
	mu         sync.RWMutex
	aliyun     *AliyunCred
	esxi       *ESXiCred
	deepseek   *DeepSeekCred
}

var global = &Store{}

func Global() *Store {
	return global
}

func (s *Store) SetAliyun(c *AliyunCred) {
	s.mu.Lock()
	defer s.mu.Unlock()
	s.aliyun = c
}

func (s *Store) GetAliyun() *AliyunCred {
	s.mu.RLock()
	defer s.mu.RUnlock()
	return s.aliyun
}

func (s *Store) SetESXi(c *ESXiCred) {
	s.mu.Lock()
	defer s.mu.Unlock()
	s.esxi = c
}

func (s *Store) GetESXi() *ESXiCred {
	s.mu.RLock()
	defer s.mu.RUnlock()
	return s.esxi
}

func (s *Store) SetDeepSeek(c *DeepSeekCred) {
	s.mu.Lock()
	defer s.mu.Unlock()
	s.deepseek = c
}

func (s *Store) GetDeepSeek() *DeepSeekCred {
	s.mu.RLock()
	defer s.mu.RUnlock()
	return s.deepseek
}

func (s *Store) HasAliyun() bool {
	s.mu.RLock()
	defer s.mu.RUnlock()
	return s.aliyun != nil && s.aliyun.AccessKeyID != "" && s.aliyun.AccessKeySecret != ""
}

func (s *Store) HasESXi() bool {
	s.mu.RLock()
	defer s.mu.RUnlock()
	return s.esxi != nil && s.esxi.URL != "" && s.esxi.User != "" && s.esxi.Password != ""
}

func (s *Store) HasDeepSeek() bool {
	s.mu.RLock()
	defer s.mu.RUnlock()
	return s.deepseek != nil && s.deepseek.APIKey != ""
}

func (s *Store) Clear() {
	s.mu.Lock()
	defer s.mu.Unlock()
	s.aliyun = nil
	s.esxi = nil
	s.deepseek = nil
}

File: src/credential.rs
//! 凭证仅保存在内存中（App 持有的字段），绝不写入磁盘。
//! 进程退出后内存自然释放，无持久化痕迹。

#[derive(Debug, Clone, Default)]
pub struct AliyunCred {
    pub access_key_id: String,
    pub access_key_secret: String,
}

#[derive(Debug, Clone, Default)]
pub struct EsxiCred {
    pub url: String,
    pub user: String,
    pub password: String,
}

#[derive(Debug, Clone, Default)]
pub struct DeepSeekCred {
    pub api_key: String,
}

#[derive(Debug, Clone, Default)]
pub struct Credentials {
    pub aliyun: Option<AliyunCred>,
    pub esxi: Option<EsxiCred>,
    pub deepseek: Option<DeepSeekCred>,
}

impl Credentials {
    pub fn has_aliyun(&self) -> bool {
        self.aliyun
            .as_ref()
            .is_some_and(|c| !c.access_key_id.is_empty() && !c.access_key_secret.is_empty())
    }

    pub fn has_esxi(&self) -> bool {
        self.esxi
            .as_ref()
            .is_some_and(|c| !c.url.is_empty() && !c.user.is_empty() && !c.password.is_empty())
    }

    pub fn has_deepseek(&self) -> bool {
        self.deepseek.as_ref().is_some_and(|c| !c.api_key.is_empty())
    }
}
